src/main/java/Deadline.java

```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task {
+    private final String time;
+    private static final String type = "[D]";
+
+    public Deadline(String name, String time) {
+        super(name);
+        this.time = time;
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + time + ")";
```

Comment:
Excellent formatting here, but one suggestion would be that since type is defined as "[D]" here already, would it be better to return type + super.toString() + ...?